internal/catalogmetadata/client/client.go

```diff
@@ -121,5 +128,31 @@ func PopulateExtraFields(catalogName string, channels []*catalogmetadata.Channel
 		}
 	}
 
+	// if package is deprecated, add deprecation to bundle
+	// if channel is deprecated, add deprecation to bundle
+	// if bundle is deprecated, add deprecation to bundle
+	for i := range bundles {
+		for _, dep := range deprecations {
+			if bundles[i].Package == dep.Package {
+				for _, entry := range dep.Entries {
+					switch entry.Reference.Schema {
+					case declcfg.SchemaPackage:
+						bundles[i].Deprecations = append(bundles[i].Deprecations, entry)
+					case declcfg.SchemaChannel:
+						for _, ch := range bundles[i].InChannels {
+							if ch.Name == entry.Reference.Name {
+								bundles[i].Deprecations = append(bundles[i].Deprecations, entry)
+							}
+						}
+					case declcfg.SchemaBundle:
+						if bundles[i].Name == entry.Reference.Name {
+							bundles[i].Deprecations = append(bundles[i].Deprecations, entry)
+						}
+					}
+				}
+			}
+		}
+	}
```

Comment:
This is very deeply nested. I'll look into ways to have less nesting here.

Reply:
Linter doesn't seem to be upset about it, anyone feel strongly about this nesting such that I need to refactor it?

Reply:
Could this work? It would at least make the runtime not be `num(bundles) * num(deprecations)`
```suggestion
	deprecationMap := make(map[string]*catalogmetadata.Deprecation, 0)
	for _, deprecation := range deprecations {
		deprecationMap[deprecation.Package] = deprecation
	}

	for i := range bundles {
		if deprecation, ok := deprecationMap[bundles[i].Package]; ok {
			for _, entry := range deprecation.Entries {
				switch entry.Reference.Schema {
				case declcfg.SchemaPackage:
					bundles[i].Deprecations = append(bundles[i].Deprecations, entry)
				case declcfg.SchemaChannel:
					for _, ch := range bundles[i].InChannels {
						if ch.Name == entry.Reference.Name {
							bundles[i].Deprecations = append(bundles[i].Deprecations, entry)
						}
					}
				case declcfg.SchemaBundle:
					if bundles[i].Name == entry.Reference.Name {
						bundles[i].Deprecations = append(bundles[i].Deprecations, entry)
					}
				}
			}
		}
	}
```
Probably fine to leave the channels and entries loops in there though.

Reply:
Oh, but if you can have multiple deprecation entries with the same package name that wouldn't work, would it?

Reply:
Updated in the latest force push (force pushed to squash commits)